src/precice/config/Configuration.cpp

```diff
@@ -64,10 +59,6 @@ void Configuration::xmlTagCallback(const xml::ConfigurationContext &context, xml
 {
   PRECICE_TRACE(tag.getName());
   if (tag.getName() == "precice-configuration") {
-    _dimensions = tag.getIntAttributeValue("dimensions");
-    _dataConfiguration->setDimensions(_dimensions);
-    _meshConfiguration->setDimensions(_dimensions);
-    _participantConfiguration->setDimensions(_dimensions);
```

Comment:
We only set the mesh configuration automatically, everything else gets the dimensions from the meshes.